rosa-hypershift/rosa-hosted-wrapper.py

```diff
@@ -828,22 +850,58 @@ def _wait_for_workers(kubeconfig, worker_nodes, wait_time, cluster_name):
-            for condition in conditions:
-                if 'type' in condition and condition['type'] == 'Ready':
-                    status.append(condition['status'])
-        status_list = {i: status.count(i) for i in status}
-        ready_nodes = status_list['True'] if 'True' in status_list else 0
+
+        # First we find nodes which label nodePool match the machinepool name and then we check if type:Ready is on the conditions
+        ready_nodes = sum(
+            len(list(filter(lambda x: x.get('type') == 'Ready' and x.get('status') == "True", node['status']['conditions'])))
+            for node in nodes
+            if node.get('metadata', {}).get('labels', {}).get('hypershift.openshift.io/nodePool')
+            and machinepool_name in node['metadata']['labels']['hypershift.openshift.io/nodePool']
+        ) if nodes else 0
+
         if ready_nodes == worker_nodes:
-            logging.info("Found %d Ready nodes on cluster %s. Expected: %d. Stopping wait." % (ready_nodes, cluster_name, worker_nodes))
-            return ready_nodes
+            logging.info("Found %d/%d ready nodes on machinepool %s for cluster %s. Stopping wait." % (ready_nodes, worker_nodes, machinepool_name, cluster_name))
+            result.append(ready_nodes)
+            result.append(str(datetime.datetime.utcnow().timestamp() - starting_time))
+            return result
         else:
-            logging.info("Found %d Ready nodes on cluster %s. Expected: %d. Waiting 15 seconds for next check..." % (ready_nodes, cluster_name, worker_nodes))
+            logging.info("Found %d/%d ready nodes on machinepool %s for cluster %s. Waiting 15 seconds for next check..." % (ready_nodes, worker_nodes, machinepool_name, cluster_name))
             time.sleep(15)
-    logging.error("Waiting time expired. After %d minutes there are %d ready nodes (Expected: %d) on cluster %s" % (wait_time, ready_nodes, worker_nodes, cluster_name))
-    return ready_nodes
+    logging.error("Waiting time expired. After %d minutes there are %d/%d ready nodes on %s machinepool for cluster %s" % (wait_time, ready_nodes, worker_nodes, machinepool_name, cluster_name))
+    result.append(ready_nodes)
+    result.append("")
+    return result
+
+
+def _add_machinepools(rosa_cmnd, kubeconfig, metadata, machinepool_name, machinepool_flavour, machinepool_labels, machinepool_taints, machinepool_replicas):
+    logging.info('Creating %d machinepools %s-ID on %s, one per AWS Zone' % (len(metadata['zones']), machinepool_name, metadata['cluster_name']))
+    machines_per_zone = machinepool_replicas // len(metadata['zones'])
+    extra_machines = machinepool_replicas % len(metadata['zones'])
+    zone_machines = [machines_per_zone] * len(metadata['zones'])
+    if extra_machines > 0:
+        zone_machines[-1] += extra_machines
+    for id, zone in enumerate(metadata['zones']):
+        machinepool_cmd = [rosa_cmnd, "create", "machinepool",
```

Comment:
don't we need to pass subnets for this now? or does zone pick an existing on if it exists already?

Reply:
No, it only needs the name of the AZ. It must be getting the information about the subnet from the cluster